content/security/docs/spark/secrets.md

```diff
@@ -0,0 +1,187 @@
+
+### Deploy Secrets Store CSI Drivers and AWS Secrets and Configuration Provider
+
+
+#### Secrets Store CSI Drivers
+
+Configure EKS Cluster with `Secrets Store CSI Driver`. 
+
+```
+helm repo add secrets-store-csi-driver \
+  https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts
+
+helm install -n kube-system csi-secrets-store \
+  --set syncSecret.enabled=true \
+  --set enableSecretRotation=true \
+  secrets-store-csi-driver/secrets-store-csi-driver
+
+```
+
+Deploy the `AWS Secrets and Configuration Provider` to use AWS Secrets Manager
+
+#### AWS Secrets and Configuration Provider
+
+```
+kubectl apply -f https://raw.githubusercontent.com/aws/secrets-store-csi-driver-provider-aws/main/deployment/aws-provider-installer.yaml
+```
+
+### Define the `SecretProviderClass`
+
+The `SecretProviderClass` is how you present present your secret in Kubernetes, below you find a definition of a `SecretProviderClass`. 
+There are few parameters that are important:
+
+- The `provider` must be set to `aws`.
+- The `objectName` must be the name of the secret you want to use as defined in AWS. 
+Here the secret is called `db-creds`.
+- The `objectType` must be set to `secretsmanager`.
+
+```
+cat > db-cred.yaml << EOF
+
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: mysql-spark-secret
+spec:
+  provider: aws
+  parameters:
+    objects: |
+        - objectName: "db-creds"
+          objectType: "secretsmanager"
+EOF
+```
+
+```
+kubectl apply -f db-cred.yaml
+```
+In the terminal apply the above command to create `SecretProviderClass`, 
+The `kubectl` command must be executed in the namespace of the where your job is going to be executed. 
+
+### Pod Template
+
+In the executor podtemplate you should define it as follow to mount the secret. The example below show how you can define it.
+There are few points that are important to mount the secret:
+
+- `secretProviderClass`: this should have the same name as the one define above. In this case it is `mysql-spark-secret`.
+- `mountPath`: Is where the secret is going to be available to the pod. In this example it will be in `/var/secrets`
+When defining the `mountPath` make sure you do not specify the ones reserved by EMR on EKS as defined [here](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/pod-templates.html). 
+
+```
+apiVersion: v1
+kind: Pod
+
+spec:
+  containers:
+    - name: spark-kubernetes-executors
+      volumeMounts:
+        - mountPath: "/var/secrets"
+          name: mysql-cred
+          readOnly: true
+  volumes:
+      - name: mysql-cred
+        csi:
+          driver: secrets-store.csi.k8s.io
+          readOnly: true
+          volumeAttributes:
+            secretProviderClass: mysql-spark-secret
+```
+
+This podtemplate must be uploaded to S3 and referenced in the job submit command as shown below.
+
```

Comment:
I'd add some info around making sure to have network path to RDS database and to configure RDS Security Group to allow IP subnet range of Spark application pods.